Solved issue about right locale

umpire274 opened the pull request:
This solution solve the problem about 'gettext' command that not load right translation instead the default (en).

mivek commented on metar_taf_parser/commons/i18n.py:
```diff
@@ -1,11 +1,18 @@
+langAvailable = os.listdir(localedir)
+defLocale = locale.getdefaultlocale()
+langCounty = defLocale[0]
+lang = langCounty[:2]
```
```suggestion
lang = locale.getdefaultlocale()[0][:2]
```

mivek commented:
Build is failing because forked repositories do not have access to environment variable. 
I added one suggestion otherwise the PR is ok

umpire274 commented:
Ok. I'll make the update on Monday.

Thanks,
Alex.

Il sab 31 lug 2021, 16:57 Jean-Kevin KPADEY ***@***.***> ha
scritto:

> Build is failing because forked repositories do not have access to
> environment variable.
> I added one suggestion otherwise the PR is ok
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mivek/python-metar-taf-parser/pull/17#issuecomment-890360462>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTBV2YFJ4B3HNONYIJTPTT2QFNTANCNFSM5BGNMCWA>
> .
>

umpire274 commented:
With the last commit I've implemented the management of 'Runway State group'.

Sometimes in Metar string compare the runway state for the airport like in these examples:

```
ULMM 020730Z 14002MPS 9000 -SHRA BKN017CB 15/14 Q0997 R13/190065 NOSIG RMK QFE741
ULMM 021330Z 28002MPS 7000 -SHRA SCT009 BKN015CB 13/13 Q1001 R31/290160 NOSIG RMK OBST OBSC QFE743

UUDD 020730Z 22003MPS 8000 -SHRA SCT029CB 22/19 Q1010 R88/290042 TEMPO -TSRA
UUDD 021330Z 13005MPS CAVOK 30/18 Q1005 R14R/290042 R14L/CLRD60 NOSIG

UHMA 020800Z 19002MPS CAVOK 09/05 Q1007 R01/29//60 NOSIG RMK TDZ R01/19002MPS END R01/22001G02MPS QFE751/1002

```
The strings for runway state are positioned after Pressure string (for example R01/29//60): to decode these strings I used the page [This is how you read a METAR - section Runway conditions](https://metar-taf.com/explanation)

I've also updated the messages.pot with new string and updated the english translation and italian translation.

I'm here for any suggestions.

Alex.

mivek commented:
Hello @umpire274 
You can revert the last commit, the runway condition will be implemented in the same way it is in the Java version of the project. I haven't had the time to do it yet. 
This PR is only about the locale.

umpire274 commented:
Ciao Kevin.

Python: I've reverted to version without Runway State Group as you requested.
I've another question. I saw you have a PHP version for MetarTafParser but the last commit is 2 years old. Can I try to update that version? I've just forked.
Can you tell me what update (major or minor) I have to check into Java version? Sorry, but I'm not familiar with Java.
I need some hints to know what updates need to be done to align the PHP version with Java.

Alex.
On ago 2 2021, at 7:30 pm, Jean-Kevin KPADEY ***@***.***> wrote:
>
>
> Hello @umpire274 (https://github.com/umpire274)
> You can revert the last commit, the runway condition will be implemented in the same way it is in the Java version of the project. I haven't had the time to do it yet.
> This PR is only about the locale.
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (https://github.com/mivek/python-metar-taf-parser/pull/17#issuecomment-891204252), or unsubscribe (https://github.com/notifications/unsubscribe-auth/AACTBV5B3OXQUZAH6NQ4WBTT23I2ZANCNFSM5BGNMCWA).
>

mivek commented:
Hello @umpire274 

I do not think I'll work on the PHP version of the project. I wanted to archive the repository so I won't be fixing any issue. It takes a lot of time to maintain all projects.

The java version of the project is the original one and the most advanced. It currently works on Java 8 and later.
I then started the python project and then the PHP one. All 3 projects are designed the same way, the classes have the same name and the methods/functions do the same thing.

If you really want to upgrade the PHP version I suggest you go back to the Java version 2 years ago. Maybe it would be better to start your own or use an already existing parser.
